src/pages/[guild]/members.tsx

```diff
@@ -141,17 +141,21 @@ const GuildPage = (): JSX.Element => {
   )
 
   useEffect(() => {
-    /**
-     * Using native browser api, so it doesn't cause a rerender & trigger the general
-     * loading bar set up in _app.tsx. The downside is, if we filter for something,
-     * navigate to another page then click the back button, it won't work. We could
-     * add some logic to the _app.tsx effect to filter out query-only changes
-     * (https://stackoverflow.com/questions/62368109/update-router-query-without-firing-page-change-event-in-next-js),
-     * but I'm not sure the performance cost would worth handling this edge case, so
-     * left it this way for now
-     */
-    window.history.pushState("", "", `?${queryString}`)
-  }, [queryString])
+    if (!router.isReady) return
+
+    const searchParams = new URLSearchParams(queryString)
+    const query = {}
+    searchParams.forEach((value, key) => {
```

Comment:
Why do we have to rebuild the query? If it's just because it's returned stringified, we can change that by removing the `.toString` in `buildQueryStringFromState` and renaming it to `buildQueryFromState`, then applying `.toString` just when passing it to `useMembers`

Reply:
It was because `router.replace` requires a query object, where keys are the names of the query parameters and values are the corresponding values. URLSearchParams is not compatible with this format.

Also, I didn't realise that we only use `buildQueryStringFromState` in this component. So maybe this query building logic could be moved there instead.